### PR TITLE
[Naitei][Gem City-state] Dynamic form bill

### PR DESCRIPTION
**Pull request by SuKay712**
## Related Tickets
https://edu-redmine.sun-asterisk.vn/issues/79544

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)

https://github.com/user-attachments/assets/e8af5d40-8a17-4502-bee6-75688c32c511

### CODE REVIEW

**Comment by SuKay712** [comment]
ready

File: app/controllers/bills_controller.rb [hunk 1]
```diff
@@ -45,4 +79,67 @@ def find_bill
...
+
+    @session = Stripe::Checkout::Session.create({
+      payment_method_types: %w(card), # rubocop:disable Layout/FirstHashElementIndentation
+      line_items: @line_items,
+      metadata: @metadata,
+      payment_method_options: {card: {request_three_d_secure: "any"}},
+      mode: "payment",
+      success_url: bills_url,
+      cancel_url: bills_url
+                                                })
```
Comment (ThaiKhacDuoc) [inline app/controllers/bills_controller.rb]: indent xa v hả huynh

File: app/controllers/bills_controller.rb [hunk 2]
```diff
@@ -45,4 +79,67 @@ def find_bill
...
+  end
+
+  def load_lines_item cart_details
+    @line_items = cart_details.map do |detail|
+      {
+        quantity: detail.quantity,
+        price_data: {
+          currency: Settings.money_unit,
+          unit_amount:
+            (detail.product.price * Settings.dollar_convert * 100).to_i,
```
Comment (ThaiKhacDuoc) [inline app/controllers/bills_controller.rb]: Đưa vào setting nha huynh
`100`

**Review by ThaiKhacDuoc — COMMENTED** [review]
cmt

File: app/controllers/bills_controller.rb (hunk 1 above)
Comment (SuKay712) [reply app/controllers/bills_controller.rb]: để z đúng indent rùi mà :v

File: app/views/bills/create.html.erb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    var stripe = Stripe('<%= Figaro.env.stripe_publishable_key %>');
+
+    stripe.redirectToCheckout({
+      sessionId: '<%= @session.id %>'
+    }).then(function (result) {
+      if (result.error) {
+        console.error(result.error.message);
+      }
+    });
+  });
```
Comment (ghost) [inline app/views/bills/create.html.erb:12]: Đưa vào file js đi em

**Review by ghost — COMMENTED** [review]
cmt

**Comment by SuKay712** [comment]
ready

File: app/controllers/bills_controller.rb [hunk 4]
```diff
@@ -45,4 +79,68 @@ def find_bill
...
+
+    @session = Stripe::Checkout::Session.create({
+      payment_method_types: %w(card), # rubocop:disable Layout/FirstHashElementIndentation
+      line_items: @line_items,
+      metadata: @metadata,
+      payment_method_options: {card: {request_three_d_secure: "any"}},
+      mode: "payment",
+      success_url: bills_url,
+      cancel_url: bills_url
+                                                })
```
Comment (ghost) [inline app/controllers/bills_controller.rb]:
```suggestion
    @session = Stripe::Checkout::Session.create({
                           payment_method_types: %w(card), # rubocop:disable Layout/FirstHashElementIndentation
                           line_items: @line_items,
                           metadata: @metadata,
                           payment_method_options: {card: {request_three_d_secure: "any"}},
                           mode: "payment",
                           success_url: bills_url,
                           cancel_url: bills_url
                         })
```
Như này mới đúng chứ

**Review by ghost — COMMENTED** [review]
cmt

File: app/controllers/bills_controller.rb (hunk 4 above)
Comment (SuKay712) [reply app/controllers/bills_controller.rb]: dạ rubocop nó bắt là ({ với }) thẳng hàng ạ

**Comment by SuKay712** [comment]
ready

**Review by ghost — APPROVED** [review]

**Pull request merged**